In [2]:
import pandas as pd
import numpy as np


In [3]:
df = pd.read_excel("C:/Users/yogin/OneDrive/Documents/SLC/SEM-2/PROGRAMMING/Final Assignment/Tivo Survey Case Data(1)(1)(1).xlsx")

In [4]:
df.head() # importing the data

ID  Gender Marital Status   Work Status Education  Annual Income (x1000 $)  \
0   1    male        married  professional      none                       49   
1   2    male         single          none      none                       46   
2   3    male        married  professional        BA                       58   
3   4    male        married          none       PhD                       51   
4   5  female         single          none      none                       46   

   Age       Location Purchasing Decision-maker        Purchasing Location  \
0   30        Florida                    family  mass-consumer electronics   
1   36        Alabama                    single  mass-consumer electronics   
2   66  Massachusetts                    family           specialty stores   
3   78       New York                    family  mass-consumer electronics   
4   52        Montana                    single  mass-consumer electronics   

   Monthly Electronics Spend  Monthly Household Spend  \
0                         35                      150   
1                         35                      163   
2                         64                      103   
3                         33                      154   
4                         45                      161   

   Purchasing Frequency (every x months) Technology Adoption  \
0                                     13                late   
1                                     26                late   
2                                     13               early   
3                                     22                late   
4                                     47                late   

   TV Viewing (hours/day)                            Favorite feature  
0                       2  saving favorite shows to watch as a family  
1                      10  saving favorite shows to watch as a family  
2                       0                               time shifting  
3                       5  saving favorite shows to watch as a family  
4                       2  saving favorite shows to watch as a family

Q2(a)-Insert a column labeled “Annual spending on electronics” that calculates the
annual spending on electronics.


In [5]:
df['Annual spending on electronics'] = df['Monthly Electronics Spend'] * 12

In [6]:
df.head()

ID  Gender Marital Status   Work Status Education  Annual Income (x1000 $)  \
0   1    male        married  professional      none                       49   
1   2    male         single          none      none                       46   
2   3    male        married  professional        BA                       58   
3   4    male        married          none       PhD                       51   
4   5  female         single          none      none                       46   

   Age       Location Purchasing Decision-maker        Purchasing Location  \
0   30        Florida                    family  mass-consumer electronics   
1   36        Alabama                    single  mass-consumer electronics   
2   66  Massachusetts                    family           specialty stores   
3   78       New York                    family  mass-consumer electronics   
4   52        Montana                    single  mass-consumer electronics   

   Monthly Electronics Spend  Monthly Household Spend  \
0                         35                      150   
1                         35                      163   
2                         64                      103   
3                         33                      154   
4                         45                      161   

   Purchasing Frequency (every x months) Technology Adoption  \
0                                     13                late   
1                                     26                late   
2                                     13               early   
3                                     22                late   
4                                     47                late   

   TV Viewing (hours/day)                            Favorite feature  \
0                       2  saving favorite shows to watch as a family   
1                      10  saving favorite shows to watch as a family   
2                       0                               time shifting   
3                       5  saving favorite shows to watch as a family   
4                       2  saving favorite shows to watch as a family   

   Annual spending on electronics  
0                             420  
1                             420  
2                             768  
3                             396  
4                             540

Q2(b) Insert another column labeled “Spending as % of income” that calculates annual spending on electronics as a percentage of annual income.

In [7]:
df['Spending as % of income'] = round((df['Annual spending on electronics'] / (df['Annual Income (x1000 $)']*1000)) * 100,2)

In [8]:
df.head()

ID  Gender Marital Status   Work Status Education  Annual Income (x1000 $)  \
0   1    male        married  professional      none                       49   
1   2    male         single          none      none                       46   
2   3    male        married  professional        BA                       58   
3   4    male        married          none       PhD                       51   
4   5  female         single          none      none                       46   

   Age       Location Purchasing Decision-maker        Purchasing Location  \
0   30        Florida                    family  mass-consumer electronics   
1   36        Alabama                    single  mass-consumer electronics   
2   66  Massachusetts                    family           specialty stores   
3   78       New York                    family  mass-consumer electronics   
4   52        Montana                    single  mass-consumer electronics   

   Monthly Electronics Spend  Monthly Household Spend  \
0                         35                      150   
1                         35                      163   
2                         64                      103   
3                         33                      154   
4                         45                      161   

   Purchasing Frequency (every x months) Technology Adoption  \
0                                     13                late   
1                                     26                late   
2                                     13               early   
3                                     22                late   
4                                     47                late   

   TV Viewing (hours/day)                            Favorite feature  \
0                       2  saving favorite shows to watch as a family   
1                      10  saving favorite shows to watch as a family   
2                       0                               time shifting   
3                       5  saving favorite shows to watch as a family   
4                       2  saving favorite shows to watch as a family   

   Annual spending on electronics  Spending as % of income  
0                             420                     0.86  
1                             420                     0.91  
2                             768                     1.32  
3                             396                     0.78  
4                             540                     1.17

Q3.	Create a table for each attribute (e.g., Gender) and record the percentage of responses for each answer (e.g., % of respondents answering "male" and % answering “female”).



In [9]:
gender = df['Gender'].value_counts(normalize = True)*100
gender

male      53.5
female    46.5
Name: Gender, dtype: float64

Q4.	Create a correlation matrix for the data set and describe the relationship you observe between the following variables:

a.	Age vs. Purchasing Frequency

In [10]:
correlation_matrix = round(df.corr(),4)
correlation_matrix

C:\Users\yogin\AppData\Local\Temp\ipykernel_28048\1216383806.py:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  correlation_matrix = round(df.corr(),4)


ID  Annual Income (x1000 $)  \
ID                                     1.0000                   0.0107   
Annual Income (x1000 $)                0.0107                   1.0000   
Age                                   -0.0415                   0.1325   
Monthly Electronics Spend             -0.0274                   0.3610   
Monthly Household Spend               -0.0549                   0.2572   
Purchasing Frequency (every x months)  0.0245                  -0.1264   
TV Viewing (hours/day)                -0.0104                   0.0843   
Annual spending on electronics        -0.0274                   0.3610   
Spending as % of income               -0.0089                   0.0404   

                                          Age  Monthly Electronics Spend  \
ID                                    -0.0415                    -0.0274   
Annual Income (x1000 $)                0.1325                     0.3610   
Age                                    1.0000                    -0.0449   
Monthly Electronics Spend             -0.0449                     1.0000   
Monthly Household Spend               -0.0126                     0.8038   
Purchasing Frequency (every x months) -0.0013                    -0.2779   
TV Viewing (hours/day)                -0.0360                     0.2472   
Annual spending on electronics        -0.0449                     1.0000   
Spending as % of income               -0.3028                     0.8200   

                                       Monthly Household Spend  \
ID                                                     -0.0549   
Annual Income (x1000 $)                                 0.2572   
Age                                                    -0.0126   
Monthly Electronics Spend                               0.8038   
Monthly Household Spend                                 1.0000   
Purchasing Frequency (every x months)                  -0.2338   
TV Viewing (hours/day)                                  0.1916   
Annual spending on electronics                          0.8038   
Spending as % of income                                 0.6520   

                                       Purchasing Frequency (every x months)  \
ID                                                                    0.0245   
Annual Income (x1000 $)                                              -0.1264   
Age                                                                  -0.0013   
Monthly Electronics Spend                                            -0.2779   
Monthly Household Spend                                              -0.2338   
Purchasing Frequency (every x months)                                 1.0000   
TV Viewing (hours/day)                                                0.1635   
Annual spending on electronics                                       -0.2779   
Spending as % of income                                              -0.1833   

                                       TV Viewing (hours/day)  \
ID                                                    -0.0104   
Annual Income (x1000 $)                                0.0843   
Age                                                   -0.0360   
Monthly Electronics Spend                              0.2472   
Monthly Household Spend                                0.1916   
Purchasing Frequency (every x months)                  0.1635   
TV Viewing (hours/day)                                 1.0000   
Annual spending on electronics                         0.2472   
Spending as % of income                                0.2358   

                                       Annual spending on electronics  \
ID                                                            -0.0274   
Annual Income (x1000 $)                                        0.3610   
Age                                                           -0.0449   
Monthly Electronics Spend                                      1.0000   
Monthly Household Spend                                        

In [11]:
correlation_Age_Purchasing = correlation_matrix.loc['Age', 'Purchasing Frequency (every x months)']
correlation_Age_Purchasing

-0.0013

Ans:- There is negative correlation between Age & purchase frequency

Q4(b) Age vs. TV Viewing

In [12]:
correlation_age_TV = correlation_matrix.loc['Age', 'TV Viewing (hours/day)']
correlation_age_TV

-0.036

Ans:- There is negative correlation between Age & TV viewing

Q4(c)Annual Income vs. TV Viewing

In [13]:
correlation_Income_TV = correlation_matrix.loc['Annual Income (x1000 $)', 'TV Viewing (hours/day)']
correlation_Income_TV

0.0843

Ans:- There is positive correlation between Income & TV viewing

Q4(d) Monthly Electronic Spending vs Monthly Household Spending

In [14]:
correlation_Elec_Household = correlation_matrix.loc['Monthly Electronics Spend', 'Monthly Household Spend']
correlation_Elec_Household

0.8038

Ans:- There is positive correlation between Income & TV viewing

Q4(e) e.	Are there any other relationships from the matrix are any high enough to make one of the attributes redundant? In other words, you can explain any of the attributes by simply saying "it behaves like attribute X?”

## "annual spending as a percentage of income" has a high positive correlation with "monthly electronics spend," of 0.82 it suggests that as individuals or households allocate a larger proportion of their annual income to spending, a significant portion of that spending is directed towards electronics.


Q5(a)How many married men who are early adopters have monthly electronics spend high enough that they can afford to purchase a TiVo for $499 and still be able to spend more on electronics in the next two years? 

In [15]:
filtered_df = df[(df['Gender'] == 'male') & (df['Marital Status'] == 'married') & (df['Technology Adoption'] == 'early')]
filtered_df

ID Gender Marital Status   Work Status Education  \
2      3   male        married  professional        BA   
6      7   male        married  professional      none   
7      8   male        married          none      none   
8      9   male        married  professional      none   
12    13   male        married  professional        BA   
..   ...    ...            ...           ...       ...   
990  991   male        married  professional      none   
993  994   male        married  professional      none   
994  995   male        married  professional        BA   
996  997   male        married  professional        MA   
997  998   male        married  professional      none   

     Annual Income (x1000 $)  Age       Location Purchasing Decision-maker  \
2                         58   66  Massachusetts                    family   
6                         33   62     California                    single   
7                         29   30  New Hampshire                    single   
8                         57   60  Massachusetts                    family   
12                        64   80     New Jersey                    family   
..                       ...  ...            ...                       ...   
990                       34   42         Hawaii                    family   
993                       31   79  New Hampshire                    single   
994                       53   38       Maryland                    family   
996                       32   73          Maine                    single   
997                       28   65        Georgia                    single   

    Purchasing Location  Monthly Electronics Spend  Monthly Household Spend  \
2      specialty stores                         64                      103   
6              discount                         18                       40   
7                retail                         23                       75   
8      specialty stores                         74                      358   
12     specialty stores                         53                      257   
..                  ...                        ...                      ...   
990              retail                         20                       47   
993            discount                         12                       54   
994    specialty stores                         40                      123   
996              retail                         15                       73   
997            discount                         10                       29   

     Purchasing Frequency (every x months) Technology Adoption  \
2                                       13               early   
6                                       41               early   
7                                        9               early   
8                                        1               early   
12                                      10               early   
..                                     ...                 ...   
990                                     26               early   
993                                     38               early   
994                                     11               early   
996                                     21               early   
997                                     11               early   

     TV Viewing (hours/day)  Favorite feature  Annual spending on electronics  \
2                         0     time shifting                             768   
6                         0       cool gadget                             216   
7                         1  schedule control                             276   
8                         0  schedule control                             888   
12                        1       cool gadget                             636   
..                      ...               ...                             ...   
990                       2  schedule control                  

In [16]:
filtered_df['Total Electronics Spend Next 2 Years'] = filtered_df['Monthly Electronics Spend'] * 24
filtered_df

C:\Users\yogin\AppData\Local\Temp\ipykernel_28048\1992747699.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_df['Total Electronics Spend Next 2 Years'] = filtered_df['Monthly Electronics Spend'] * 24


ID Gender Marital Status   Work Status Education  \
2      3   male        married  professional        BA   
6      7   male        married  professional      none   
7      8   male        married          none      none   
8      9   male        married  professional      none   
12    13   male        married  professional        BA   
..   ...    ...            ...           ...       ...   
990  991   male        married  professional      none   
993  994   male        married  professional      none   
994  995   male        married  professional        BA   
996  997   male        married  professional        MA   
997  998   male        married  professional      none   

     Annual Income (x1000 $)  Age       Location Purchasing Decision-maker  \
2                         58   66  Massachusetts                    family   
6                         33   62     California                    single   
7                         29   30  New Hampshire                    single   
8                         57   60  Massachusetts                    family   
12                        64   80     New Jersey                    family   
..                       ...  ...            ...                       ...   
990                       34   42         Hawaii                    family   
993                       31   79  New Hampshire                    single   
994                       53   38       Maryland                    family   
996                       32   73          Maine                    single   
997                       28   65        Georgia                    single   

    Purchasing Location  Monthly Electronics Spend  Monthly Household Spend  \
2      specialty stores                         64                      103   
6              discount                         18                       40   
7                retail                         23                       75   
8      specialty stores                         74                      358   
12     specialty stores                         53                      257   
..                  ...                        ...                      ...   
990              retail                         20                       47   
993            discount                         12                       54   
994    specialty stores                         40                      123   
996              retail                         15                       73   
997            discount                         10                       29   

     Purchasing Frequency (every x months) Technology Adoption  \
2                                       13               early   
6                                       41               early   
7                                        9               early   
8                                        1               early   
12                                      10               early   
..                                     ...                 ...   
990                                     26               early   
993                                     38               early   
994                                     11               early   
996                                     21               early   
997                                     11               early   

     TV Viewing (hours/day)  Favorite feature  Annual spending on electronics  \
2                         0     time shifting                             768   
6                         0       cool gadget                             216   
7                         1  schedule control                             276   
8                         0  schedule control                             888   
12                        1       cool gadget                             636   
..                      ...               ...                             ...   
990                       2  schedule control                  

In [17]:
can_afford_tivo = len(filtered_df[filtered_df['Total Electronics Spend Next 2 Years'] > 499])
can_afford_tivo

134

Ans 134 married male with  early adopter to tech are having electricity expenses for 2 years more than 499

Q5(b)How many women with an education level of MA or PhD are making purchasing decisions for electronics without discussing them with a spouse, either because they are single or because they are making purchasing decisions without the involvement of their spouses? 

In [18]:
filtered_df_f = df[(df['Gender'] == 'female') & (df['Education'].isin(['MA', 'PhD']))]

In [19]:
independent = len(filtered_df_f[(filtered_df_f['Marital Status'] == 'single') | (filtered_df_f['Purchasing Decision-maker'] == 'single')])
independent

56

Ans total 56 female 

Q5(c)How many early adopters purchase electronics at least once every year, and do so in stores that specialize in electronics? 

In [20]:
filtered_df_adp = df[df['Technology Adoption'] == 'early']

In [21]:
frequent_shoppers = filtered_df_adp[(filtered_df_adp['Purchasing Frequency (every x months)'] >= 12) & (filtered_df_adp['Purchasing Location'] == 'specialty stores')]
frequent_shoppers

ID  Gender Marital Status   Work Status Education  \
2      3    male        married  professional        BA   
41    42  female        married  professional        BA   
46    47  female        married  professional        BA   
72    73  female        married  professional      none   
125  126    male        married  professional        BA   
138  139    male        married  professional      none   
203  204    male        married  professional        MA   
207  208    male        married  professional        MA   
224  225    male        married  professional        MA   
242  243  female        married  professional        BA   
244  245    male         single  professional        BA   
246  247  female        married  professional        MA   
253  254  female         single  professional        MA   
293  294  female         single  professional       PhD   
305  306  female        married  professional        MA   
334  335  female         single  professional        BA   
342  343    male        married  professional      none   
348  349    male         single  professional      none   
361  362    male        married  professional        MA   
367  368  female        married  professional        MA   
385  386    male        married  professional        BA   
393  394    male        married  professional        BA   
399  400    male        married  professional        MA   
410  411  female         single  professional        BA   
440  441    male        married  professional        BA   
446  447    male        married  professional        BA   
469  470    male        married  professional        BA   
478  479    male        married  professional        BA   
496  497  female        married  professional        BA   
499  500    male        married  professional        MA   
507  508    male        married  professional      none   
512  513    male        married  professional      none   
561  562    male        married  professional        BA   
564  565    male         single  professional      none   
596  597    male        married  professional        BA   
619  620  female        married  professional        BA   
636  637    male        married  professional      none   
639  640  female        married  professional        BA   
674  675    male        married  professional        BA   
772  773  female         single  professional        BA   
783  784    male        married  professional        BA   
785  786    male        married  professional        MA   
797  798    male         single  professional        BA   
831  832    male        married  professional        BA   
834  835    male         single  professional       PhD   
836  837    male         single  professional        BA   
858  859    male        married  professional        MA   
884  885    male         single  professional        BA   
899  900    male         single  professional        BA   
902  903  female         single  professional        BA   
941  942  female        married  professional        MA   

     Annual Income (x1000 $)  Age       Location Purchasing Decision-maker  \
2                         58   66  Massachusetts                    family   
41                        54   43        Vermont                    family   
46                        59   63        Vermont                    family   
72                        53   65      Tennessee                    family   
125                       55   46         Hawaii                    family   
138                       49   22     Washington                    family   
203                       55   50      Tennessee                    family   
207                       45   24        Vermont                    family   
224                       58   67         Hawaii                    family   
242                       55   23  New Hampshire                    family   
244                       57   68       New York                    single   
2

In [22]:
len(frequent_shoppers)

51

Ans 51 customers

Q5(d)	How many seniors (over the age of 65) spend more than six hours a day watching TV? What is their income range? What is their average annual income? 

In [23]:
seniors_watching_tv = df[(df['Age'] > 65) & (df['TV Viewing (hours/day)'] > 6)]
seniors_watching_tv.head()


ID  Gender Marital Status   Work Status Education  \
20    21  female        married          none        MA   
74    75  female        married          none      none   
101  102  female         single  professional        MA   
115  116  female        married          none      none   
157  158  female        married  professional       PhD   

     Annual Income (x1000 $)  Age   Location Purchasing Decision-maker  \
20                        50   75   Colorado                    family   
74                        55   73   Delaware                    family   
101                       46   79   Missouri                    single   
115                       54   73  Tennessee                    family   
157                       51   78    Arizona                    family   

           Purchasing Location  Monthly Electronics Spend  \
20   mass-consumer electronics                         47   
74   mass-consumer electronics                         44   
101  mass-consumer electronics                         41   
115  mass-consumer electronics                         51   
157  mass-consumer electronics                         35   

     Monthly Household Spend  Purchasing Frequency (every x months)  \
20                       172                                     17   
74                       208                                     40   
101                      173                                     43   
115                      221                                     37   
157                      162                                     13   

    Technology Adoption  TV Viewing (hours/day)  \
20                 late                       8   
74                 late                       7   
101                late                      13   
115                late                      11   
157                late                      12   

                               Favorite feature  \
20   saving favorite shows to watch as a family   
74   saving favorite shows to watch as a family   
101  saving favorite shows to watch as a family   
115  saving favorite shows to watch as a family   
157  saving favorite shows to watch as a family   

     Annual spending on electronics  Spending as % of income  
20                              564                     1.13  
74                              528                     0.96  
101                             492                     1.07  
115                             612                     1.13  
157                             420                     0.82

In [24]:

income_min = seniors_watching_tv['Annual Income (x1000 $)'].min()
income_min


41

In [25]:
income_max = seniors_watching_tv['Annual Income (x1000 $)'].max()
income_max


55

In [26]:
average_income = seniors_watching_tv['Annual Income (x1000 $)'].mean()
average_income


48.6

In [27]:
len(seniors_watching_tv)

20

Q6.	Create a gropuby for each of the following variables listed below that will show the variable and annual spend in the format of “.describe()” function. 
a.	Gender with Monthly Electronic Spend
b.	Martial Status with Monthly Electronic Spend
c.	Education with Monthly Electronic Spend


Q6(a)	Gender with Monthly Electronic Spend

In [28]:
gender_group_by = df.groupby('Gender')['Monthly Electronics Spend'].describe()
gender_group_by


count       mean        std  min   25%   50%   75%   max
Gender                                                          
female  465.0  28.776344  15.120261  7.0  17.0  23.0  38.0  83.0
male    535.0  32.861682  18.368460  9.0  17.0  30.0  45.0  88.0

Q6(b) Martial Status with Monthly Electronic Spend

In [29]:
marital_status_group_by = df.groupby('Marital Status')['Monthly Electronics Spend'].describe()
marital_status_group_by

count       mean        std  min   25%   50%   75%   max
Marital Status                                                          
married         720.0  30.704167  17.491424  9.0  17.0  21.0  44.0  85.0
single          280.0  31.625000  15.868594  7.0  18.0  32.0  36.0  88.0

Q6(c)Education with Monthly Electronic Spend

In [30]:
education_group_edu = df.groupby('Education')['Monthly Electronics Spend'].describe()
education_group_edu

count       mean        std   min    25%   50%   75%   max
Education                                                            
BA         260.0  40.242308  18.711149  12.0  20.00  39.5  54.0  85.0
MA         124.0  40.370968  14.678931  13.0  32.00  38.0  47.0  88.0
PhD        110.0  38.290909  12.102275  12.0  32.25  37.0  43.0  85.0
none       506.0  22.294466  12.583749   7.0  15.00  18.0  22.0  83.0

Q7 7.	Based on your analysis of the dataset, who do you believe the ideal customer group  who marketing should focus their attention on? 

In [31]:
average_income = df['Annual Income (x1000 $)'].mean()

In [32]:
filtered_df_analysis = df[(df['Annual Income (x1000 $)'] > average_income) &
                 (df['Purchasing Frequency (every x months)'] > 12) &
                 (df['Technology Adoption'] == 'early') &
                 (df['Age'] < 50)]

In [33]:
filtered_df_analysis

ID  Gender Marital Status   Work Status Education  \
125  126    male        married  professional        BA   
138  139    male        married  professional      none   
207  208    male        married  professional        MA   
242  243  female        married  professional        BA   
246  247  female        married  professional        MA   
253  254  female         single  professional        MA   
293  294  female         single  professional       PhD   
334  335  female         single  professional        BA   
478  479    male        married  professional        BA   
496  497  female        married  professional        BA   
499  500    male        married  professional        MA   
512  513    male        married  professional      none   
619  620  female        married  professional        BA   
785  786    male        married  professional        MA   
834  835    male         single  professional       PhD   
884  885    male         single  professional        BA   
941  942  female        married  professional        MA   

     Annual Income (x1000 $)  Age       Location Purchasing Decision-maker  \
125                       55   46         Hawaii                    family   
138                       49   22     Washington                    family   
207                       45   24        Vermont                    family   
242                       55   23  New Hampshire                    family   
246                       55   38          Idaho                    family   
253                       55   33   North Dakota                    single   
293                       52   25       Virginia                    single   
334                       53   39    Mississippi                    single   
478                       53   40     California                    family   
496                       52   42    Connecticut                    family   
499                       55   46        Arizona                    family   
512                       49   24      Tennessee                    family   
619                       56   29        Arizona                    family   
785                       50   26        Georgia                    family   
834                       54   47        Georgia                    single   
884                       52   45   Pennsylvania                    single   
941                       55   37       Illinois                    family   

    Purchasing Location  Monthly Electronics Spend  Monthly Household Spend  \
125    specialty stores                         50                      225   
138    specialty stores                         51                      166   
207    specialty stores                         65                      211   
242    specialty stores                         61                      258   
246    specialty stores                         38                      115   
253    specialty stores                         82                      165   
293    specialty stores                         25                       31   
334    specialty stores                         39                       69   
478    specialty stores                         65                      180   
496    specialty stores                         57                      250   
499    specialty stores                         72                      252   
512    specialty stores                         49                      158   
619    specialty stores                         51                      225   
785    specialty stores                         49                      119   
834    specialty stores                         46                      104   
884    specialty stores                         62                      145   
941    specialty stores                         77                      308   

     Purchasing Frequency (every x months) Technology Adoption  \
125                                     13       

To focus on a specific group of customers based on the criteria  mentioned above,  we can do so using data filtering techniques to get more business. Here's a general approach to achieve this:

Filter by Income: Identify customers whose income is greater than the average income of all customers.

Filter by Purchase Frequency: Customers who have a purchase frequency of more than 12.

Filter by Technology Adoption: Early adopters of technology.

Filter by Age: Finally, from this group, we will select only those customers who are less than 50 years old.
